yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnAllocatorSuite.scala

```diff
@@ -85,6 +87,7 @@ class YarnAllocatorSuite extends SparkFunSuite with Matchers with BeforeAndAfter
   }
 
   class MockSplitInfo(host: String) extends SplitInfo(null, host, null, 1, null) {
+    override def hashCode(): Int = Random.nextInt()
```

Comment:
This is wrong now though. `hashCode` has to be deterministic and always return the same value. There is nothing wrong with always returning 0. The problem is actually with the `equals` method, but, it won't matter here.